### PR TITLE
update update critic

### PR DESCRIPTION
**Pull request by vFf0621**

### CODE REVIEW

File: agents/alm.py [hunk 1]
```diff
@@ -293,20 +293,26 @@ def _intrinsic_reward_loss(self, z, action_batch, z_next, z_next_prior, log, met
 
     def update_critic(self, z_batch, action_batch, reward_batch, z_next_batch, discount_batch, std, log, metrics):
         with torch.no_grad():
-            z_next_batch = self.model(z_batch, action_batch).sample()
             next_action_dist = self.actor(z_next_batch, std)
             next_action_batch = next_action_dist.sample()
+
             Q1_, Q2_ = self.critic_target(z_next_batch, next_action_batch)
```
Comment (ktolnos) [inline agents/alm.py:299]: You need to use get_critic_value here as well

File: agents/alm.py [hunk 2]
```diff
@@ -293,20 +293,26 @@ def _intrinsic_reward_loss(self, z, action_batch, z_next, z_next_prior, log, met
...
 
         self.critic_opt.zero_grad()
         critic_loss.backward()
         critic_grad_norm = torch.nn.utils.clip_grad_norm_(utils.get_parameters(self.critic_list), max_norm=self.max_grad_norm, error_if_nonfinite=True)
         self.critic_opt.step()
 
+        self.critic_opt.zero_grad()
+        critic_loss.backward()
+        critic_grad_norm = torch.nn.utils.clip_grad_norm_(utils.get_parameters(self.critic_list), max_norm=self.max_grad_norm, error_if_nonfinite=True)
+        self.critic_opt.step()
```
Comment (ktolnos) [inline agents/alm.py:314]: Why is this code duplicated?

File: agents/alm.py (hunk 1 above)
Comment (vFf0621) [reply agents/alm.py:299]: Sorry sir, I realized this as well and it was fixed like 5 minutes later than the PR.

File: agents/alm.py (hunk 2 above)
Comment (vFf0621) [reply agents/alm.py:314]: Sorry sir, I realized this as well and it was fixed 5 minutes after the PR.

**Review by vFf0621 — COMMENTED** [review]
I think I’m finished sir. Do you want me to merge it?

**Pull request merged**